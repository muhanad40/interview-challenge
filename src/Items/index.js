import React from 'react';
import PropTypes from 'prop-types';

import Item from '../Item';

const Items = (props) => {
  return (
    <div className="col-4">
      <ul className="item-picker">
        {props.items.map(item => <Item key={item.id} {...item} />)}
      </ul>
    </div>
  );
};

Items.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape(Item.propTypes)
  ),
};

Items.defaultProps = {
  items: [],
};

export default Items;
