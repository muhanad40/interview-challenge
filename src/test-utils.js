import React from 'react';
import { shallow } from 'enzyme';

export function createRenderer(Component, props) {
  return (extraProps) => shallow(<Component {...props} {...extraProps} />);
}
