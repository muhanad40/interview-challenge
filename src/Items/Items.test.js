require('../test-setup');

import Items from './index';
import { createRenderer } from '../test-utils';

describe('Items', () => {
  let render;
  const initProps = {
    items: [
      {
        id: 1001,
        name: 'Kale Caesar Pasta, Turmeric Satay Broccoli & Lemon Cashew Greens',
        dietaries: ['v', 've', 'df', 'gf', 'n!'],
      },
    ],
  };

  beforeAll(() => {
    render = createRenderer(Items, initProps);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should render list of items from data prop', () => {
    const mockItems = [
      {
        id: 1002,
        name: 'Hake & Smoky Chickpeas, Brown Rice & Quinoa, Roasted Roots',
        dietaries: ['gf', 'df', 'rsf'],
      },
      {
        id: 1003,
        name: 'Dill & Swiss Chard Potato Cakes, Summer Tabbouleh & Roasted Roots',
        dietaries: ['gf', 'df', 'v', 've', 'n!'],
      },
      {
        id: 1004,
        name: 'Hake & Smoky Chickpeas, Herby Potatoes & Turmeric Satay Broccoli',
        dietaries: ['df', 'gf', 'rsf', 'n!'],
      },
      {
        id: 1005,
        name: 'Cuban Beef, Brown Rice & Quinoa, Green Pepper & Butterbean Salad',
        dietaries: ['gf', 'df'],
      },
    ];
    const component = render({
      items: mockItems,
    });

    expect(component.find('Item')).toHaveLength(mockItems.length);
  });
});
