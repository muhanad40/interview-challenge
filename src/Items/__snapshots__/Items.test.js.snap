// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items should render correctly 1`] = `
<div
  className="col-4"
>
  <ul
    className="item-picker"
  >
    <Item
      dietaries={
        Array [
          "v",
          "ve",
          "df",
          "gf",
          "n!",
        ]
      }
      id={1001}
      key="1001"
      name="Kale Caesar Pasta, Turmeric Satay Broccoli & Lemon Cashew Greens"
    />
  </ul>
</div>
`;
